### PR TITLE
Don't compare against cache if forceFetch is true

### LINKED ISSUE #1214: Refetch is not refetching
Hello! 

I have a pretty simple use case here where I have a query which fetches all contacts for a particular account. In another area of the app, I have an import feature which allows the user to import a csv of contacts. Upon returning to the contacts page, I call refetch on the query, however, the new data is never actually fetched, only the old data is returned without the added contacts. Upon refreshing the app, the new data is properly fetched. Am I missing something? 

```
function isLoading(props) {
  const { accounts, contacts } = props;

  return accounts.loading || contacts.loading;
}

module.exports = class extends React.Component {
  componentDidMount() {
    this.props.contacts.refetch();
  }

  render() {
    console.log(isLoading(this.props));

    const Footer = () => (
      <div className="box-footer text-right">
        <Button
          label="Create Contact"
          size="sm"
          onClick={ () => showContact() }
        />
      </div>
    );

    return !isLoading(this.props) ? (
      <WidgetBox boxTitle="Contacts" Footer={ Footer }>
        { renderContacts(this.props) }
      </WidgetBox>
    ) : null;
  }
};
```

### PR DESCRIPTION
**Pull request by stevewillard**
Fixes #1214

### CODE REVIEW

File: src/core/QueryManager.ts [hunk 1]
```diff
@@ -502,7 +502,7 @@ export class QueryManager {
 
     // If this is not a force fetch, we want to diff the query against the
     // store before we fetch it from the network interface.
-    if (!forceFetch) {
+    if (fetchType !== FetchType.refetch || !forceFetch) {
```
Comment (calebmer) [inline src/core/QueryManager.ts]: I think it’s correct here to use `&&` and not `||`. See if that fixes the tests.

**Comment by stevewillard** [comment]
Ok - I'll try it out later tonight or tomorrow. Also, any idea why I can't sign the CLA? I keep getting a 500 error when I fill out all the fields.

**Comment by stubailo** [comment]
@stevewillard I just changed some stuff about CLA let me check

File: src/core/QueryManager.ts (hunk 1 above)
Comment (stevewillard) [inline src/core/QueryManager.ts]: Thanks -- seems to be passing now. I merged/squashed the commits.

**Comment by calebmer** [comment]
Looks like that fixed it. @stevewillard would you be willing to write a test to prevent future regressions? Otherwise it looks good to me 👍

**Comment by stevewillard** [comment]
Hey, sorry about lagging on this PR. I am really swamped with work. I don't think I'll have time for a few weeks to finish this up. Happy to close it out if someone wants to take another stab at it.

**Comment by calebmer** [comment]
I’m comfortable merging this without a test if it is going to be too much effort. @helfer what do you think?

**Comment by helfer** [comment]
Yeah, that works for me, especially since we might replace `forceFetch` and `noFetch` with an explicit fetch policy (more on that soon). I'll merge it now! 🙂 Thanks @stevewillard!

**Pull request merged**